main.go

```diff
@@ -54,6 +53,20 @@ func (s *stringSlice) Set(value string) error {
 	return nil
 }
 
+func defaultStateDirectory() string {
+	//  pam_systemd sets XDG_RUNTIME_DIR but not other dirs.
+	xdgDataHome := os.Getenv("XDG_DATA_HOME")
```

Comment:
Any reasons behind not using `os/user.User`? :thinking:

Reply:
https://github.com/moby/buildkit/issues/333